tests/tracking/test_rest_tracking.py

```diff
@@ -579,7 +576,11 @@ def test_validate_path_is_safe_bad(path):
         r".\..\path",
         r"path\..\to\file",
         r"path\..\..\to\file",
-        # Drive-relative absolute paths
+        # Drive-relative paths
+        r"C:path",
+        r"C:path/",
+        r"C:path/to/file",
+        r"C:../path/to/file",
```

Comment:
I get `r"C:../path/to/file"` should be considered as bad path, but why the first three also bad? Relative path without `..` should be good right?

Reply:
Disallowing C:/path/to/file is a behavior change if it's supposed to be a safe path. Could we just disallowing '..' by splitting the path on ':' as well?

Reply:
> /mlflow-artifacts/artifacts/C:path/to/file -> for some reason will access path/to/file instead of mlartifacts/path/to/file, which is unintended (we only want access to mlartifacts.)
This does look weird, they could just remove 'C:' for accessing relative paths. Thanks for the explanation :D